src/rshim_fuse.c

```diff
@@ -1064,11 +1066,11 @@ static void rshim_fuse_rshim_ioctl(fuse_req_t req, int cmd, void *arg,
 
     if (cmd == RSHIM_IOC_WRITE) {
       pthread_mutex_lock(&bd->mutex);
-      rc = bd->write_rshim(bd, chan, offset, msg.data);
+      rc = bd->write_rshim(bd, chan, offset, msg.data, msg.data_size);
       pthread_mutex_unlock(&bd->mutex);
     } else {
       pthread_mutex_lock(&bd->mutex);
-      rc = bd->read_rshim(bd, chan, offset, &data);
+      rc = bd->read_rshim(bd, chan, offset, &data, msg.data_size);
       msg.data = data;
       pthread_mutex_unlock(&bd->mutex);
     }
```

Comment:
Once we have RSHIM_IOC_WRITE2/RSHIM_IOC_READ2, the code here will be slightly different